elfpy/data/acquire_data.py

```diff
@@ -256,6 +261,26 @@ def main(
                 if block_pool_info:
                     postgres.add_pool_infos([block_pool_info], session)
 
+                # keep querying until it returns to avoid random crashes with ValueError on some intermediate block
+                block_checkpoint_info = None
+                for _ in range(RETRY_COUNT):
+                    try:
+                        checkpoint_info_dict = hyperdrive_interface.get_hyperdrive_checkpoint_info(
+                            web3, hyperdrive_contract, block_number
+                        )
+                        # Set defaults
+                        for key in db_schema.CheckpointInfo.__annotations__:
```

Comment:
Can we move this logic outside of the try catch? The try catch should really only be over the chain query. Eventually, we should move this logic to hyperdrive_interface

Reply:
Now that I see these lines side-by-side, it seems the obvious choice would be to set defaults either in the dataclass itself or in the `add_item` classes inside postgres, where we have access to db_schema stuff no problem:

```py
for key in db_schema.CheckpointInfo.__annotations__:
    if key not in checkpoint_info_dict:
        checkpoint_info_dict[key] = None
if block_checkpoint_info := db_schema.CheckpointInfo(**checkpoint_info_dict):
    postgres.add_checkpoint_infos([block_checkpoint_info], session)
```

Reply:
I'll just move them outside of the try/except in this PR, and we can re-org them in a future PR.